WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java

```diff
@@ -661,6 +661,10 @@ private void startSiteIconUpload(final String filePath) {
         SiteModel site = getSelectedSite();
         if (site != null) {
             MediaModel media = buildMediaModel(file, site);
+            if (media == null) {
+                ToastUtils.showToast(getActivity(), R.string.file_not_found, ToastUtils.Duration.SHORT);
+                return;
```

Comment:
I think we didn't catch the exception / close the code flow before because we were hoping to find more information in the crash report. What do you think about also adding some logging we can report back to Crashlytics so we can understand why this situation happens?

Reply:
This could happen when `FluxCUtils.mediaModelFromLocalUri` returns `null`.
(Actually  we're not seeing problems in `MySiteFragment`, and added the code to be safe).

It return null when `String path = MediaUtils.getRealPathFromURI(context, uri);`  cannot complete for some reasons. I can add a log there, but don't think it would be of any help.

Reply:
The real fix of this NPE is done in f539301f797df318d537f5ddafc3ce6b53618267.
Since the code that builds the media model in `PhotoPickerActivity` expects the file to be accessible/there, but for some reason it is not, and then it dispatches an action with empty payload.

Reply:
> The real fix of this NPE is done in f539301.

Sure, I only commented on this other file/line but meant to make a general comment for every place where this PR shows a toast and end the flow. 👍

Reply:
> It returns null when `String path = MediaUtils.getRealPathFromURI(context, uri);` cannot complete for some reasons. I can add a log there, but don't think it would be of any help.

My fear here is that if they can't open a file once, they probably won't be able to ever open such a file  (that is, they will keep getting the toast each time they try). That will be silent for us, and we won't be able to help them as we don't know something happened. What if we add some sort of tracking?  Maybe not logging, but using tracks so we know if something grows? (I just wouldn't want it to go unnoticed for long time and hit the app ratings if anything happens and we can't see it) wdyt?

Reply:
Yeah, good call. Added the logging in 5c6255b